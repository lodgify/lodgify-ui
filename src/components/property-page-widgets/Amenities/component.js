import React from 'react';
import PropTypes from 'prop-types';

import { VIEW_MORE } from 'utils/default-strings';
import { Grid } from 'layout/Grid';
import { GridRow } from 'layout/GridRow';
import { GridColumn } from 'layout/GridColumn';
import { Heading } from 'typography/Heading';

import { getDefaultItems } from './utils/getDefaultItems';
import { hasExtraItems } from './utils/hasExtraItems';
import { getCategoryMarkup } from './utils/getCategoryMarkup';
import { getExtraItemsMarkup } from './utils/getExtraItemsMarkup';
import { getExtraItems } from './utils/getExtraItems';

/**
 * The standard widget for displaying the amenities of a property.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  amenities,
  hasExtraItemsInModal,
  headingText,
  isStacked,
  modalTriggerText,
}) => (
  <Grid className="is-amenities" columns={isStacked ? 1 : 3}>
    <GridRow>
      {headingText && (
        <GridColumn width={12}>
          <Heading>{headingText}</Heading>
        </GridColumn>
      )}
      {getDefaultItems(amenities, isStacked).map((amenity, index) =>
        getCategoryMarkup(amenity, index, isStacked)
      )}
      {hasExtraItems(amenities, isStacked) &&
        getExtraItemsMarkup(
          hasExtraItemsInModal,
          modalTriggerText,
          getExtraItems(amenities, isStacked)
        )}
    </GridRow>
  </Grid>
);

Component.displayName = 'Amenities';

Component.defaultProps = {
  hasExtraItemsInModal: false,
  headingText: null,
  isStacked: false,
  modalTriggerText: VIEW_MORE,
};

Component.propTypes = {
  /** The amenity categories. */
  amenities: PropTypes.arrayOf(
    PropTypes.shape({
      /** The name of the icon to display for the category.
       * [See here for the full list of valid icon names](https://github.com/lodgify/lodgify-ui/blob/master/src/components/elements/Icon/constants.js)
       */
      iconName: PropTypes.string.isRequired,
      /** The list of amenities */
      items: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
      /** The category name */
      name: PropTypes.string.isRequired,
    })
  ).isRequired,
  /** Are the extra items displayed in a modal. */
  hasExtraItemsInModal: PropTypes.bool,
  /** The text to display as a heading at the top of the amenities. */
  headingText: PropTypes.string,
  /** Are the amenities displayed stacked on top of one another */
  isStacked: PropTypes.bool,
  /** The text for the modal trigger */
  modalTriggerText: PropTypes.string,
};
