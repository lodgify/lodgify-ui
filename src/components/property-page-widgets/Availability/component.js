import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { CalendarMonth, CalendarDay } from 'react-dates';
import { BLOCKED_MODIFIER } from 'react-dates/constants';
import moment from 'moment';
import { Card } from 'semantic-ui-react';
import { size } from 'lodash';

import { buildKeyFromStrings } from 'utils/build-key-from-strings';
import { Dropdown } from 'inputs/Dropdown';
import { Grid } from 'layout/Grid';
import { GridColumn } from 'layout/GridColumn';
import { GridRow } from 'layout/GridRow';
import { Heading } from 'typography/Heading';
import { Icon, ICON_NAMES } from 'elements/Icon';
import { Paragraph } from 'typography/Paragraph';
import { withResponsive } from 'utils/with-responsive';
import {
  AVAILABILITY,
  NEXT,
  PREVIOUS,
  PROPERTIES,
  UNAVAILABLE,
  VIEW_AVAILABILITY_FOR,
} from 'utils/default-strings';

import { BLOCKED_DAY_CLASS } from './constants';
import { isDayBlockedOrBeforeCurrentDate } from './utils/isDayBlockedOrBeforeCurrentDate';
import { getMonthsToDisplay } from './utils/getMonthsToDisplay';
import { getNextStartDate } from './utils/getNextStartDate';
import { getPreviousStartDate } from './utils/getPreviousStartDate';
import { renderMonthHeader } from './utils/renderMonthHeader';

import 'react-dates/initialize';

/**
 * The standard widget for displaying the availability of a room.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
class Component extends PureComponent {
  state = {
    startDate: moment(),
  };

  componentDidMount = () => moment.locale(this.props.localeCode);

  handleClickNextMonth = () =>
    this.setState({
      startDate: getNextStartDate(
        this.state.startDate,
        this.props.isUserOnMobile
      ),
    });

  handleClickPreviousMonth = () =>
    this.setState({
      startDate: getPreviousStartDate(
        this.state.startDate,
        this.props.isUserOnMobile
      ),
    });

  renderCalendarDay = props => {
    const modifiers = new Set([]);
    const { getIsDayBlocked } = this.props;

    if (isDayBlockedOrBeforeCurrentDate(props.day, getIsDayBlocked)) {
      modifiers.add(BLOCKED_DAY_CLASS);
      modifiers.add(BLOCKED_MODIFIER);
    }

    return <CalendarDay {...props} modifiers={modifiers} />;
  };

  reloadCalendarOnRoomSelection = (evt, value) => {
    this.props.onChangeRoomDropdown(evt, value);
    this.forceUpdate();
  };

  render() {
    const { startDate } = this.state;
    const {
      headingText,
      isUserOnMobile,
      legendLabel,
      nextMonthsButtonLabel,
      previousMonthsButtonLabel,
      propertyDropdownLabel,
      propertyDropdownPlaceholderLabel,
      roomOptionsWithImages,
    } = this.props;

    return (
      <div>
        <Heading size="small">{headingText}</Heading>
        <Grid>
          <GridRow>
            {size(roomOptionsWithImages) > 0 && (
              <GridColumn
                computer={7}
                mobile={12}
                tablet={6}
                verticalAlignContent="middle"
              >
                <Grid>
                  <GridColumn
                    computer={3}
                    mobile={5}
                    tablet={12}
                    verticalAlignContent="middle"
                  >
                    <Paragraph size="tiny" weight="heavy">
                      {propertyDropdownLabel}
                    </Paragraph>
                  </GridColumn>
                  <GridColumn computer={7} mobile={7} tablet={12}>
                    <Dropdown
                      icon={ICON_NAMES.MAP_PIN}
                      label={propertyDropdownPlaceholderLabel}
                      onChange={this.reloadCalendarOnRoomSelection}
                      options={roomOptionsWithImages}
                    />
                  </GridColumn>
                </Grid>
              </GridColumn>
            )}
            <GridColumn
              floated="right"
              only="tablet computer"
              textAlign="right"
              verticalAlignContent="middle"
              width={5}
            >
              <Icon
                color="light grey"
                isLabelLeft
                labelText={legendLabel}
                name={ICON_NAMES.SQUARE}
              />
            </GridColumn>
          </GridRow>
        </Grid>
        <Card fluid>
          <Grid padded>
            <GridRow>
              {getMonthsToDisplay(startDate, isUserOnMobile).map(
                (month, index) => (
                  <GridColumn
                    className="availability-calendar-wrapper"
                    computer={6}
                    key={buildKeyFromStrings('month-column', index)}
                    mobile={12}
                    tablet={6}
                  >
                    <CalendarMonth
                      isVisible
                      month={month}
                      renderCalendarDay={this.renderCalendarDay}
                      renderMonthElement={renderMonthHeader}
                    />
                  </GridColumn>
                )
              )}
            </GridRow>
            <GridRow>
              <GridColumn width={6}>
                <Icon
                  labelText={previousMonthsButtonLabel}
                  name={ICON_NAMES.ARROW_LEFT}
                  onClick={this.handleClickPreviousMonth}
                />
              </GridColumn>
              <GridColumn textAlign="right" width={6}>
                <Icon
                  isLabelLeft
                  labelText={nextMonthsButtonLabel}
                  name={ICON_NAMES.ARROW_RIGHT}
                  onClick={this.handleClickNextMonth}
                />
              </GridColumn>
            </GridRow>
          </Grid>
        </Card>
        <Grid>
          <GridColumn only="mobile" textAlign="right" width={12}>
            <GridColumn>
              <Icon
                color="light grey"
                isLabelLeft
                labelText={legendLabel}
                name={ICON_NAMES.SQUARE}
              />
            </GridColumn>
          </GridColumn>
        </Grid>
      </div>
    );
  }
}

export const ComponentWithResponsive = withResponsive(Component);

Component.displayName = 'Availability';

Component.defaultProps = {
  getIsDayBlocked: () => false,
  headingText: AVAILABILITY,
  legendLabel: UNAVAILABLE,
  localeCode: 'en',
  nextMonthsButtonLabel: NEXT,
  onChangeRoomDropdown: Function.prototype,
  previousMonthsButtonLabel: PREVIOUS,
  propertyDropdownLabel: VIEW_AVAILABILITY_FOR,
  propertyDropdownPlaceholderLabel: PROPERTIES,
  roomOptionsWithImages: [],
};

Component.propTypes = {
  /**
   * A function that is passed to determine if the current date should be blocked, the function is passed a moment object
   * @param {Moment} momentObj
   * @returns{boolean}
   */
  getIsDayBlocked: PropTypes.func,
  /** The text to display as a heading at the top of the widget. */
  headingText: PropTypes.string,
  /**
   * Is the user on a mobile device.
   * Provided by `withResponsive` so ignored in the styleguide.
   * @ignore
   */
  isUserOnMobile: PropTypes.bool.isRequired,
  /** The label for the legend. */
  legendLabel: PropTypes.string,
  /** The ISO 639-1 locale code which changes the format and language of days of the week and the months of the year. */
  localeCode: PropTypes.string,
  /** The label for the next months button. */
  nextMonthsButtonLabel: PropTypes.string,
  /**
   * A function that triggered when a room is changed in the room dropdown
   * @param {Event} event
   * @param {String} currentRoomValue
   */
  onChangeRoomDropdown: PropTypes.func,
  /** The label for the previous months button. */
  previousMonthsButtonLabel: PropTypes.string,
  /** The label that goes next to the property dropdown. */
  propertyDropdownLabel: PropTypes.string,
  /** The label that goes inside the dropdown when there are no options. */
  propertyDropdownPlaceholderLabel: PropTypes.string,
  /** The options which the user can select for the room field. */
  roomOptionsWithImages: PropTypes.PropTypes.arrayOf(
    PropTypes.shape({
      /** A list of one or more strings separated by commas indicating a set of source sizes for the image. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
      imageSizes: PropTypes.string,
      /** A list of one or more strings separated by commas indicating a set of possible image sources for the user agent to use for the image. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
      imageSrcSet: PropTypes.string,
      /** The source url of the image to display with the option. */
      imageUrl: PropTypes.string,
      /** The visible text for the option. */
      text: PropTypes.string.isRequired,
      /** The underlying value for the option. */
      value: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.number,
        PropTypes.string,
      ]),
    })
  ),
};
