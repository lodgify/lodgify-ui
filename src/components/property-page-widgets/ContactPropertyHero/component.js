import React, { Fragment } from 'react';
import { string, bool, shape, number, oneOfType, arrayOf } from 'prop-types';

import { FullBleed } from 'media/FullBleed';
import { ContactHeader } from 'collections/ContactHeader';
import { FlexContainer } from 'layout/FlexContainer';
import { toUpper } from 'utils/to-upper';
import { Button } from 'elements/Button';
import { ICON_NAMES } from 'elements/Icon';
import { HorizontalGutters } from 'layout/HorizontalGutters';
import { Gallery } from 'media/Gallery';
import { Divider } from 'elements/Divider';
import { size } from 'utils/size';
import { VIEW_MORE_PICTURES } from 'utils/default-strings';

import { BOTTOM_OFFSET } from './constants';

/**
 * A hero displays a contact header with a gallery.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  backgroundImageHeight,
  backgroundImageSizes,
  backgroundImageSrcSet,
  backgroundImageUrl,
  backgroundImageWidth,
  bottomOffset,
  className,
  currencyOptions,
  currencyValue,
  headingHref,
  headingText,
  languageOptions,
  languageValue,
  navigationMenuItems,
  phoneNumber,
  placeholderBackgroundImageUrl,
  primaryButtonText,
  galleryImages,
  secondaryButtonText,
  currencyNoResultsText,
}) => {
  return (
    <FullBleed
      bottomOffset={bottomOffset}
      className={className}
      hasGradient
      imageHeight={backgroundImageHeight}
      imageUrl={backgroundImageUrl}
      imageWidth={backgroundImageWidth}
      placeholderImageUrl={placeholderBackgroundImageUrl}
      sizes={backgroundImageSizes}
      srcSet={backgroundImageSrcSet}
    >
      <ContactHeader
        className={className}
        currencyNoResultsText={currencyNoResultsText}
        currencyOptions={currencyOptions}
        currencyValue={currencyValue}
        headingHref={headingHref}
        headingText={headingText}
        isBackgroundFilled
        languageOptions={languageOptions}
        languageValue={languageValue}
        navigationMenuItems={navigationMenuItems}
        phoneNumber={phoneNumber}
        primaryButtonText={primaryButtonText}
      />
      {size(galleryImages) > 1 ? (
        <Fragment>
          <FlexContainer alignItems="flex-end">
            <HorizontalGutters>
              <Gallery
                images={galleryImages}
                trigger={
                  <Button
                    icon={ICON_NAMES.PLACEHOLDER}
                    isCompact
                    isPositionedRight
                    isSecondary
                  >
                    {toUpper(secondaryButtonText)}
                  </Button>
                }
              />
            </HorizontalGutters>
          </FlexContainer>
          <Divider />
        </Fragment>
      ) : null}
    </FullBleed>
  );
};

Component.displayName = 'ContactPropertyHero';

Component.defaultProps = {
  backgroundImageHeight: undefined,
  backgroundImageSizes: undefined,
  backgroundImageSrcSet: undefined,
  backgroundImageWidth: undefined,
  bottomOffset: BOTTOM_OFFSET,
  className: null,
  currencyNoResultsText: undefined,
  currencyOptions: [],
  currencyValue: null,
  headingHref: '',
  headingText: '',
  languageOptions: [],
  languageValue: null,
  phoneNumber: '',
  placeholderBackgroundImageUrl: null,
  galleryImages: [],
  secondaryButtonText: VIEW_MORE_PICTURES,
  primaryButtonText: '',
};

Component.propTypes = {
  /** The natural height of the background image. */
  backgroundImageHeight: oneOfType([string, number]),
  /** A list of one or more strings separated by commas indicating a set of source sizes for the background image. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
  backgroundImageSizes: string,
  /** A list of one or more strings separated by commas indicating a set of possible image sources for the user agent to use for the background image. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
  backgroundImageSrcSet: string,
  /** The source url of the hero's background image. */
  backgroundImageUrl: string.isRequired,
  /** The natural width of the background image. */
  backgroundImageWidth: oneOfType([string, number]),
  /** Reduce the height of the Hero with an offset, supports CSS dimensions. */
  bottomOffset: string,
  /** The custom classes. */
  className: string,
  /** The text to display when no results are returned from the currency dropdown. */
  currencyNoResultsText: string,
  /** The options which the user can select for the currency dropdown. */
  currencyOptions: arrayOf(
    shape({
      /** The visible text for the option. */
      text: string.isRequired,
      /** The underlying value for the option. */
      value: oneOfType([bool, number, string]),
    })
  ),
  /** The current value of the currency dropdown. */
  currencyValue: oneOfType([bool, number, string]),
  /** The images to display in the gallery modal. */
  galleryImages: arrayOf(
    shape({
      /** A description of the image to show above the slideshow when the image is showing. */
      descriptionText: string,
      /** A list of one or more strings separated by commas indicating a set of source sizes. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
      sizes: string,
      /** A list of one or more strings separated by commas indicating a set of possible image sources for the user agent to use. See [the MDN docs for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img). */
      srcSet: string,
      /** Title of the image to show when hovering over it on desktop browsers. */
      title: string,
      /** URL pointing to the image to display. */
      url: string.isRequired,
    })
  ),
  /** The link for the heading. */
  headingHref: string,
  /** The text for the heading. */
  headingText: string,
  /** The options which the user can select for the language dropdown. */
  languageOptions: arrayOf(
    shape({
      /** The visible text for the option. */
      text: string.isRequired,
      /** The underlying value for the option. */
      value: oneOfType([bool, number, string]),
    })
  ),
  /** The current value of the language dropdown. */
  languageValue: oneOfType([bool, number, string]),
  /** The items for a user to navigate the site. */
  navigationMenuItems: arrayOf(
    shape({
      /** The href url for an item which is a link. */
      href: string,
      /** Sub navigation items to display as a [`<Submenu />`](#submenu) under an item. */
      subItems: arrayOf(
        shape({
          href: string.isRequired,
          target: string,
          text: string.isRequired,
        })
      ),
      /** Specifies where to display the linked navigation items URL. See [MDN docs `<a />` for more](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-target). */
      target: string,
      /** The visible text for an item. */
      text: string.isRequired,
    })
  ).isRequired,
  /** The phone number to display. */
  phoneNumber: string,
  /** The ISO 639-1 locale code which changes the format and language of days of the week and the months of the year in the search bars date range picker. */
  placeholderBackgroundImageUrl: string,
  /** The text to display for the primary button. */
  primaryButtonText: string,
  /** The text to display on the secondary button at the bottom of the hero. */
  secondaryButtonText: string,
};
