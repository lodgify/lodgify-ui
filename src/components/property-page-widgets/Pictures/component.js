import React from 'react';
import PropTypes from 'prop-types';

import { EXPLORE_ALL_PICTURES, PROPERTY_PICTURES } from 'utils/default-strings';
import { getFirstSixItems } from 'utils/get-first-six-items';
import { buildKeyFromStrings } from 'utils/build-key-from-strings';
import { Grid } from 'layout/Grid';
import { GridRow } from 'layout/GridRow';
import { GridColumn } from 'layout/GridColumn';
import { ShowOn } from 'layout/ShowOn';
import { Divider } from 'elements/Divider';
import { Thumbnail } from 'media/Thumbnail';
import { Heading } from 'typography/Heading';
import { Gallery } from 'media/Gallery';
import { getGalleryHeadingMarkup } from 'utils/get-gallery-heading-markup';
import { Link } from 'elements/Link';

/**
 * The standard widget for displaying pictures of a property.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  headingText,
  linkText,
  pictures,
  propertyName,
  ratingNumber,
}) => (
  <Grid columns={3}>
    <GridColumn width={12}>
      <Heading>{headingText}</Heading>
    </GridColumn>
    <GridRow>
      {getFirstSixItems(pictures).map(
        ({ imageUrl, imageSizes, imageSrcSet, label }, index) => (
          <GridColumn key={buildKeyFromStrings(imageUrl, index)}>
            <ShowOn computer parent="div" tablet widescreen>
              <Thumbnail
                imageSizes={imageSizes}
                imageSrcSet={imageSrcSet}
                imageUrl={imageUrl}
                label={label}
                size="huge"
              />
            </ShowOn>
            <ShowOn mobile parent="div">
              <Thumbnail
                hasRoundedCorners
                imageSizes={imageSizes}
                imageSrcSet={imageSrcSet}
                imageUrl={imageUrl}
                isSquare
                label={label}
                size="huge"
              />
            </ShowOn>
            <Divider size="small" />
          </GridColumn>
        )
      )}
    </GridRow>
    <GridColumn width={12}>
      <Gallery
        heading={getGalleryHeadingMarkup(propertyName, ratingNumber)}
        images={pictures}
        trigger={<Link>{linkText}</Link>}
      />
    </GridColumn>
  </Grid>
);

Component.displayName = 'Pictures';

Component.defaultProps = {
  headingText: PROPERTY_PICTURES,
  linkText: EXPLORE_ALL_PICTURES,
  propertyName: null,
  ratingNumber: null,
};

Component.propTypes = {
  /** The text to display as a heading at the top of the widget. */
  headingText: PropTypes.string,
  /** The text to display on the link at the bottom of the widget. */
  linkText: PropTypes.string,
  /** The pictures to display. See [the `ResponsiveImage` component for valid props](#/Media/ResponsiveImage).  */
  pictures: PropTypes.arrayOf(PropTypes.object).isRequired,
  /** The name of the property to display in the gallery modal. */
  propertyName: PropTypes.string,
  /** The numeral rating for the property, out of 5 */
  ratingNumber: PropTypes.number,
};
