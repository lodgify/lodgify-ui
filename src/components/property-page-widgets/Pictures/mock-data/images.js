const imageUrl =
  'https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg?w=1024&mode=max';

export const images = [
  { imageUrl, imageUrlAlternativeText: 'Entrance', label: 'Entrance' },
  { imageUrl, imageUrlAlternativeText: 'Kitchen', label: 'Kitchen' },
  {
    imageUrl,
    imageUrlAlternativeText: 'Living room',
    label: 'Living room',
  },
  { imageUrl, imageUrlAlternativeText: 'Studio', label: 'Studio' },
  {
    imageUrl,
    imageUrlAlternativeText: 'Garden room',
    label: 'Garden room',
  },
];
