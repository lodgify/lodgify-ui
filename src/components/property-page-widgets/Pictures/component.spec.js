import React from 'react';
import { mount } from 'enzyme';
import { expectComponentToHaveDisplayName } from '@lodgify/enzyme-jest-expect-helpers';

import { images } from './mock-data/images';
import { Component as Pictures } from './component';

const getPictures = () =>
  mount(<Pictures galleryImages={images} thumbnailImages={images} />);

describe('<Pictures />', () => {
  it('should render the correct structure', () => {
    const actual = getPictures();

    expect(actual).toMatchSnapshot();
  });

  it('should have `displayName` `Pictures`', () => {
    expectComponentToHaveDisplayName(Pictures, 'Pictures');
  });
});
