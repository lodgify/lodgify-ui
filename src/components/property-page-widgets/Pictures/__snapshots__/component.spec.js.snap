// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pictures /> should render the correct structure 1`] = `
<Grid
  areColumnsCentered={false}
  id={null}
  isStackable={false}
>
  <GridColumn
    verticalAlignContent="top"
    width={12}
  >
    <Heading
      className={null}
      hasMargin={true}
      isColorInverted={false}
      size="medium"
      textAlign={null}
    >
      Property pictures
    </Heading>
  </GridColumn>
  <GridRow
    horizontalAlignContent="left"
  >
    <GridColumn
      computer={4}
      data-testid="pictures-thumbnail_0"
      key="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg0"
      mobile={6}
      onClick={[Function]}
      tablet={4}
      verticalAlignContent="top"
      width={null}
    >
      <ShowOn
        computer={true}
        mobile={false}
        parent="div"
        parentProps={Object {}}
        tablet={true}
        widescreen={true}
      >
        <Thumbnail
          className="gallery-thumbnail"
          hasMargin={false}
          hasRoundedCorners={false}
          imageUrl="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
          imageUrlAlternativeText="Two cats"
          isCircular={false}
          isSquare={false}
          label="Two cats"
          size="huge"
        />
      </ShowOn>
      <ShowOn
        computer={false}
        mobile={true}
        parent="div"
        parentProps={Object {}}
        tablet={false}
        widescreen={false}
      >
        <Thumbnail
          className="gallery-thumbnail"
          hasMargin={false}
          hasRoundedCorners={true}
          imageUrl="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
          imageUrlAlternativeText="Two cats"
          isCircular={false}
          isSquare={false}
          label="Two cats"
          size="small"
        />
      </ShowOn>
      <Divider
        className=""
        hasLine={false}
        size="medium"
      />
    </GridColumn>
    <GridColumn
      computer={4}
      data-testid="pictures-thumbnail_1"
      key="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg1"
      mobile={6}
      onClick={[Function]}
      tablet={4}
      verticalAlignContent="top"
      width={null}
    >
      <ShowOn
        computer={true}
        mobile={false}
        parent="div"
        parentProps={Object {}}
        tablet={true}
        widescreen={true}
      >
        <Thumbnail
          className="gallery-thumbnail"
          hasMargin={false}
          hasRoundedCorners={false}
          imageUrl="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
          imageUrlAlternativeText="Two more cats"
          isCircular={false}
          isSquare={false}
          label="Two more cats"
          size="huge"
        />
      </ShowOn>
      <ShowOn
        computer={false}
        mobile={true}
        parent="div"
        parentProps={Object {}}
        tablet={false}
        widescreen={false}
      >
        <Thumbnail
          className="gallery-thumbnail"
          hasMargin={false}
          hasRoundedCorners={true}
          imageUrl="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
          imageUrlAlternativeText="Two more cats"
          isCircular={false}
          isSquare={false}
          label="Two more cats"
          size="small"
        />
      </ShowOn>
      <Divider
        className=""
        hasLine={false}
        size="medium"
      />
    </GridColumn>
    <GridColumn
      computer={4}
      data-testid="pictures-thumbnail_2"
      key="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg2"
      mobile={6}
      onClick={[Function]}
      tablet={4}
      verticalAlignContent="top"
      width={null}
    >
      <ShowOn
        computer={true}
        mobile={false}
        parent="div"
        parentProps={Object {}}
        tablet={true}
        widescreen={true}
      >
        <Thumbnail
          className="gallery-thumbnail"
          hasMargin={false}
          hasRoundedCorners={false}
          imageUrl="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
          imageUrlAlternativeText="Much cats"
          isCircular={false}
          isSquare={false}
          label="Much cats"
          size="huge"
        />
      </ShowOn>
      <ShowOn
        computer={false}
        mobile={true}
        parent="div"
        parentProps={Object {}}
        tablet={false}
        widescreen={false}
      >
        <Thumbnail
          className="gallery-thumbnail"
          hasMargin={false}
          hasRoundedCorners={true}
          imageUrl="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
          imageUrlAlternativeText="Much cats"
          isCircular={false}
          isSquare={false}
          label="Much cats"
          size="small"
        />
      </ShowOn>
      <Divider
        className=""
        hasLine={false}
        size="medium"
      />
    </GridColumn>
    <GridColumn
      computer={4}
      data-testid="pictures-thumbnail_3"
      key="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg3"
      mobile={6}
      onClick={[Function]}
      tablet={4}
      verticalAlignContent="top"
      width={null}
    >
      <ShowOn
        computer={true}
        mobile={false}
        parent="div"
        parentProps={Object {}}
        tablet={true}
        widescreen={true}
      >
        <Thumbnail
          className="gallery-thumbnail"
          hasMargin={false}
          hasRoundedCorners={false}
          imageUrl="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
          imageUrlAlternativeText="Two cats"
          isCircular={false}
          isSquare={false}
          label="Two cats"
          size="huge"
        />
      </ShowOn>
      <ShowOn
        computer={false}
        mobile={true}
        parent="div"
        parentProps={Object {}}
        tablet={false}
        widescreen={false}
      >
        <Thumbnail
          className="gallery-thumbnail"
          hasMargin={false}
          hasRoundedCorners={true}
          imageUrl="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
          imageUrlAlternativeText="Two cats"
          isCircular={false}
          isSquare={false}
          label="Two cats"
          size="small"
        />
      </ShowOn>
      <Divider
        className=""
        hasLine={false}
        size="medium"
      />
    </GridColumn>
    <GridColumn
      computer={4}
      data-testid="pictures-thumbnail_4"
      key="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg4"
      mobile={6}
      onClick={[Function]}
      tablet={4}
      verticalAlignContent="top"
      width={null}
    >
      <ShowOn
        computer={true}
        mobile={false}
        parent="div"
        parentProps={Object {}}
        tablet={true}
        widescreen={true}
      >
        <Thumbnail
          className="gallery-thumbnail"
          hasMargin={false}
          hasRoundedCorners={false}
          imageUrl="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
          imageUrlAlternativeText="Two more cats"
          isCircular={false}
          isSquare={false}
          label="Two more cats"
          size="huge"
        />
      </ShowOn>
      <ShowOn
        computer={false}
        mobile={true}
        parent="div"
        parentProps={Object {}}
        tablet={false}
        widescreen={false}
      >
        <Thumbnail
          className="gallery-thumbnail"
          hasMargin={false}
          hasRoundedCorners={true}
          imageUrl="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
          imageUrlAlternativeText="Two more cats"
          isCircular={false}
          isSquare={false}
          label="Two more cats"
          size="small"
        />
      </ShowOn>
      <Divider
        className=""
        hasLine={false}
        size="medium"
      />
    </GridColumn>
    <GridColumn
      computer={4}
      data-testid="pictures-thumbnail_5"
      key="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg5"
      mobile={6}
      onClick={[Function]}
      tablet={4}
      verticalAlignContent="top"
      width={null}
    >
      <ShowOn
        computer={true}
        mobile={false}
        parent="div"
        parentProps={Object {}}
        tablet={true}
        widescreen={true}
      >
        <Thumbnail
          className="gallery-thumbnail"
          hasMargin={false}
          hasRoundedCorners={false}
          imageUrl="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
          imageUrlAlternativeText="Much cats"
          isCircular={false}
          isSquare={false}
          label="Much cats"
          size="huge"
        />
      </ShowOn>
      <ShowOn
        computer={false}
        mobile={true}
        parent="div"
        parentProps={Object {}}
        tablet={false}
        widescreen={false}
      >
        <Thumbnail
          className="gallery-thumbnail"
          hasMargin={false}
          hasRoundedCorners={true}
          imageUrl="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
          imageUrlAlternativeText="Much cats"
          isCircular={false}
          isSquare={false}
          label="Much cats"
          size="small"
        />
      </ShowOn>
      <Divider
        className=""
        hasLine={false}
        size="medium"
      />
    </GridColumn>
  </GridRow>
  <GridColumn
    verticalAlignContent="top"
    width={12}
  >
    <Gallery
      data-testid="pictures-gallery"
      images={
        Array [
          Object {
            "descriptionText": "Two cats",
            "url": "https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg",
          },
          Object {
            "descriptionText": "Two more cats",
            "url": "https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg",
          },
          Object {
            "descriptionText": "Much cats",
            "url": "https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg",
          },
          Object {
            "descriptionText": "Two cats",
            "url": "https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg",
          },
          Object {
            "descriptionText": "Two more cats",
            "url": "https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg",
          },
          Object {
            "descriptionText": "Much cats",
            "url": "https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg",
          },
          Object {
            "descriptionText": "Two cats",
            "url": "https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg",
          },
          Object {
            "descriptionText": "Two more cats",
            "url": "https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg",
          },
          Object {
            "descriptionText": "Much cats",
            "url": "https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg",
          },
        ]
      }
      isOpen={false}
      onClick={[Function]}
      onClose={[Function]}
      startIndex={2}
      trigger={
        <Link
          data-testid="pictures-triggerLink"
          href={null}
          isFluid={false}
          isPositionedRight={false}
          onClick={[Function]}
          willOpenInNewTab={false}
        >
          Explore all pictures
        </Link>
      }
    />
  </GridColumn>
</Grid>
`;
